src/NBitcoin/Stealth/StealthMetadata.cs

```diff
@@ -52,7 +52,7 @@ public StealthMetadata(Key ephemKey, uint nonce)
         static TxNullDataTemplate _Template = new TxNullDataTemplate(1024 * 4);
         private static bool Fill(StealthMetadata output, Script metadata)
         {
-            var datas = _Template.ExtractScriptPubKeyParameters(metadata);
+            var datas = _Template.ExtractScriptPubKeyParameters(Network.Main, metadata);
```

Comment:
Why only Main?

Reply:
This is a stealth address so if it becomes needed the implementer can just fix it.